src/samplebuffer.h

```diff
@@ -0,0 +1,73 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h" // CSAMPLE
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+// The public interface closely follows that of std::vector.
+//
+// No resize operation is provided intentionally for maximum efficiency!
+// If the size of an existing sample buffer needs to be altered after
+// construction this can simply be achieved by swapping the contents with
+// a temporary sample buffer that has been constructed with the desired
+// size:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer tempBuffer(newSize)
+//     ... copy data from sampleBuffer to tempBuffer...
+//     sampleBuffer.swap(sampleBuffer);
+//
+// The local variable tempBuffer can be omitted if no data needs to be
+// copied from the existing sampleBuffer:
+//
+//     SampleBuffer sampleBuffer(oldSize);
+//     ...
+//     SampleBuffer(newSize).swap(sampleBuffer);
+//
+// After construction the content of the buffer is uninitialized.
+class SampleBuffer {
+    Q_DISABLE_COPY(SampleBuffer);
+
+  public:
+    typedef CSAMPLE value_type;
+    typedef int size_type;
```

Comment:
If I write "1.0f" this is equivalent to "CSAMPLE_PEAK". But the latter is more expressive even if you don't see the value anymore. It is actually a benefit that the concrete value is hidden behind the constant's name. It is good practice to give names to values. So why not to types?

We as developers have to provide these abstractions, giving names to concepts: values, types, functions, algorithms, ... An IDE will never be able to decide what "1.0f" somewhere in the code actually means: peak sample value? max. gain value? ...? Keeping complex (software) systems manageable is all about abstractions.

Another argument: The common width of "int" is still 32-bit even on 64-bit architectures. Are you sure that this will be enough for all time considering multi-channel audio at 192 kHz? We are doomed if we ever need to revert this decision later! On the other hand if you have spent a typedef and consequently used it right from the beginning? Well, nothing easier than that :)

I will remove all typedefs from SampleBuffer.